nncf/openvino/quantization/weights_compression.py

```diff
@@ -0,0 +1,125 @@
+
+        for const_port_id in allowed_metatypes_to_const_port[metatype]:
+            weight_node = get_operation_const_op(node, const_port_id)
+            if weight_node is None:
+                continue
+
+            weight_output = weight_node.output(0)
+            weight_name = weight_node.get_friendly_name()
+            original_weight_dtype = weight_output.get_element_type().to_dtype()
+            target_inputs = weight_output.get_target_inputs()
+
+            if original_weight_dtype not in [np.float32, np.float16]:
+                continue
+
+            weight = get_const_value(weight_node)
+            axes = _get_reduction_axes(metatype, node, const_port_id)
+            min_values = np.min(weight, axis=axes, keepdims=True)
+            max_values = np.max(weight, axis=axes, keepdims=True)
+            stats = OVMinMaxTensorStatistic(min_values, max_values)
+            fq_params = get_fq_params(stats)
+
+            input_low = fq_params.input_low
+            input_high = fq_params.input_high
+            assert np.allclose(fq_params.output_low, input_low)
+            assert np.allclose(fq_params.output_high, input_high)
+
+            levels = fq_params.levels
+            new_output_low = -levels // 2
+            new_output_high = levels - 1 + new_output_low
+            scale, zero_point = calculate_scale_zero_point(
+                input_low, input_high, new_output_low, new_output_high, narrow_range=False
+            )
+
+            int8_weight = np.round(weight / scale + zero_point).astype(np.int8)
+            quantized_weight = opset.constant(int8_weight, dtype=np.int8, name=weight_name)
+            convert = opset.convert(quantized_weight, original_weight_dtype)
+            sub = opset.subtract(convert, zero_point.astype(original_weight_dtype))
+            fq_name = f"{node.get_friendly_name()}/fq_weights_{const_port_id}"
+            mul = opset.multiply(sub, scale.astype(original_weight_dtype), name=fq_name)
+
+            for target_input in target_inputs:
+                target_input.replace_source_output(mul.output(0))
+
+
+def _get_reduction_axes(metatype: Type[OperatorMetatype], node: ov.Node, weight_port_id: int) -> Union[int, Tuple[int]]:
+    """
+    Determines reduction axes by given metatype and node information.
+
+    :param metatype: The metatype of the operator.
+    :param node: The OpenVINO node.
+    :param weight_port_id: The weight port ID.
+
+    :return: The reduction axes as an integer or a tuple of integers.
+    """
+    if metatype is OVMatMulMetatype:
+        transpose = node.get_attributes()[f"transpose_{'a' if weight_port_id == 0 else 'b'}"]
+        ndims = node.input(weight_port_id).get_partial_shape().rank.get_max_length()
+        channel_axes = get_matmul_channel_axes(weight_port_id, ndims, transpose)
+        axes = tuple(i for i in range(ndims) if i not in channel_axes)
+    elif metatype is OVEmbeddingMetatype:
+        axes = (metatype.const_channel_axis[0] + 1) % 2
```

Comment:
Could you explain these calculations?

Reply:
To align with torch https://github.com/openvinotoolkit/nncf/blob/c027c8b43c4865d46b8de01d8350dd338ec5a874/nncf/torch/quantization/weights_compression.py#L82-L83